File: engine/src/main/java/com/terralite/engine/physics/Velocity.java
package com.terralite.engine.physics;

public record Velocity(double x, double y, double z) {
    public static final Velocity ZERO = new Velocity(0.0, 0.0, 0.0);

    public Velocity scale(double scalar) {
        return new Velocity(x * scalar, y * scalar, z * scalar);
    }
}

File: engine/src/commonMain/kotlin/com/terralite/engine/physics/Velocity.kt
package com.terralite.engine.physics

@JvmRecord
data class Velocity(val x: Double, val y: Double, val z: Double) {
    companion object {
        @JvmField val ZERO = Velocity(0.0, 0.0, 0.0)
    }

    fun scale(scalar: Double): Velocity = Velocity(x * scalar, y * scalar, z * scalar)
}
